testsupport/wait/host.go

```diff
@@ -1744,3 +1744,77 @@ func UntilSpaceBindingHasSpaceRole(expected string) SpaceBindingWaitCriterion {
 		},
 	}
 }
+
+type SocialEventWaitCriterion struct {
+	Match func(*toolchainv1alpha1.SocialEvent) bool
+	Diff  func(*toolchainv1alpha1.SocialEvent) string
+}
+
+func matchSocialEventWaitCriterion(actual *toolchainv1alpha1.SocialEvent, criteria ...SocialEventWaitCriterion) bool {
+	for _, c := range criteria {
+		if !c.Match(actual) {
+			return false
+		}
+	}
+	return true
+}
+
+// WaitForSocialEvent waits until the SocialEvent is available with the provided criteria, if any
+func (a *HostAwaitility) WaitForSocialEvent(name string, criteria ...SocialEventWaitCriterion) (*toolchainv1alpha1.SocialEvent, error) {
+	var se *toolchainv1alpha1.SocialEvent
+
+	err := wait.Poll(a.RetryInterval, 2*a.Timeout, func() (done bool, err error) {
+		// retrieve the SocialEvent from the host namespace
+		seList := &toolchainv1alpha1.SocialEventList{}
+		if err = a.Client.List(context.TODO(), seList, client.InNamespace(a.Namespace)); err != nil {
```

Comment:
ah, you don't use the name here at all - you just list all of them. But this is not a robust approach - what if there are SocialEvents that were created in previous tests?

Reply:
yeah, let me fix that, I didn't notice this bug

Reply:
this was clearly a combination of poor copy-paste and lack of attention 😬
addressed in 75d3f6d